modules/nlu/src/backend/api.ts

```diff
@@ -118,6 +118,64 @@ export const registerRouter = async (bp: typeof sdk, app: NLUApplication) => {
       return mapError({ botId, lang, error }, res)
     }
   })
+
+  addTrainingServiceRoutes(router, app, mapError)
+}
+
+const trainingStatusSchema = Joi.object().keys({
+  status: Joi.string()
+    .valid(['idle', 'done', 'needs-training', 'training-pending', 'training', 'canceled', 'errored'])
+    .required(),
+  progress: Joi.number()
+    .default(0)
+    .min(0)
+    .max(1)
+    .required()
+})
+
+const addTrainingServiceRoutes = (
+  router: sdk.http.RouterExtension,
+  app: NLUApplication,
+  errorMapper: (err: { botId: string; lang: string; error: Error }, res: Response) => Response
+) => {
+  const repo = app.repository
+  const SUBROUTE_NAME = 'trainrepo'
+
+  router.get(`/${SUBROUTE_NAME}/trainings`, async (req, res) => {
+    const trainings = await repo.getAll()
+    res.status(200).json(trainings)
+  })
+
+  router.use(`/${SUBROUTE_NAME}/trainings/:botId/:lang`, async (req, res, next) => {
+    const { botId, lang } = req.params
+    try {
+      const training = await repo.get({ botId, language: lang })
+      if (!training) {
+        throw new Error('Training not found')
+      }
+      res.locals.training = training
+    } catch (error) {
+      return errorMapper({ botId, lang, error }, res)
+    }
+    next()
+  })
+
+  router.get(`/${SUBROUTE_NAME}/trainings/:botId/:lang`, async (req, res) => {
+    const trainings = res.locals.training as TrainingSession[]
```

Comment:
How does this work? It's some kind a cache right? Which handler will be called first? this one or the one on top?